paasta_tools/setup_marathon_job.py

```diff
@@ -733,12 +775,10 @@ def main() -> None:
     # Setting up transparent cache for http API calls
     requests_cache.install_cache("setup_marathon_jobs", backend="memory")
 
-    marathon_config = get_main_marathon_config()
-    client = marathon_tools.get_marathon_client(
-        marathon_config.get_url(), marathon_config.get_username(),
-        marathon_config.get_password(),
-    )
-    marathon_apps = marathon_tools.get_all_marathon_apps(client, embed_tasks=True)
+    system_paasta_config = load_system_paasta_config()
+    clients = marathon_tools.get_marathon_clients(marathon_tools.get_marathon_servers(system_paasta_config))
+    unique_clients = clients.get_all_clients()
+    marathon_apps_with_clients = marathon_tools.get_marathon_apps_with_clients(unique_clients, embed_tasks=True)
```

Comment:
I'm kinda worried about this part of the code scaling up 10x

Reply:
We'll see, I guess. If it's bad, we can improve it by having it only get the tasks from the relevant clients for the services in question. (We'd also want to make a similar change in `deployd.workers.PaastaDeployWorker.process_service_instance`.)